lua/cli_flash.lua

```diff
@@ -541,221 +541,254 @@ MODE_TEST_CLI = 12
+    -- tParserCommand:option('--t --plugin_type', 'plugin type'):target('strPluginType')
+    tParserCommand:option('-t', 'plugin type'):target('strPluginType')
 end
 
+function addJtagKhzArg(tParserCommand)
+    tParserCommand:option('--jtag_khz', 'JTAG clock in kHz'):target('iJtagKhz')
+end
 
--- Checks during argument parsing: Return an error if 
--- - the mode name is unknown
--- - an argument key ("-b") is unknown
--- - an argument key is known but neither required nor optional for the given mode
+function addJtagResetArg(tParserCommand)
+    tOption = tParserCommand:option('--jtag_reset',
+            'JTAG reset method. Possible values are: hard (default), soft, attach'):target('strJtagReset')
+    tOption._options = {"HardReset", "SoftReset", "Attach" }
+end
 
--- After argument parsing: Check if all required args are present 
--- If a required arg is not specified but has a default value, set the default.
--- If it does not have a default, return an error.
+local argparse = require 'argparse'
 
-function parseArgs()
-	local fOk
-	local strMsg
-	
-	local aArgs
-	local strMode
-	local tModeArgs
+local strEpilog = [==[
+Limitations:
 
-	local iArg
-	local nArgs
-	
-	fOk = true
-	aArgs = {}
-	nArgs = #arg
-	
-	-- First arg is the mode.
-	-- If no args are given, set help mode
-	if nArgs == 0 then
-		aArgs.iMode = MODE_HELP
-		
-	else
-		strMode = arg[1]
-		tModeArgs = atModeArgs[strMode]
-		
-		if tModeArgs == nil then
-			fOk = false
-			strMsg = string.format("Unknown mode: %s", strMode)
-		else
-			aArgs.iMode = tModeArgs.mode
-			
-			iArg = 2
-			-- Parse the arguments.
-			-- The last argument may be the file name and has no key.
-			while (iArg <= nArgs and fOk == true) do
-				if iArg == nArgs then
-					fOk, strMsg = parseArg(aArgs, strMode, tModeArgs, "", arg[iArg])
-					iArg = iArg + 1
-				else
-					fOk, strMsg = parseArg(aArgs, strMode, tModeArgs, arg[iArg], arg[iArg+1])
-					iArg = iArg + 2
-				end
-			end
-			
-			-- check required arguments
-			if fOk == true then
-				fOk, strMsg = checkRequiredArgs(aArgs, tModeArgs.required_args)
-			end
+The reset_netx command currently supports only the netx 90.
+
+The hash and verify_hash commands do not support the netx 90 and netIOL.
+
+
+Examples:
+
+Write file to serial flash:
+lua cli_flash.lua flash -b 1 NETX100-BSL.bin
+
+Erase boot cookie from serial flash:
+lua cli_flash.lua erase -b 1 -l 4
+
+Erase boot cookie from parallel flash:
+lua cli_flash.lua erase -b 0 -l 4
+]==]
+
+local tParser = argparse('Cli Flasher', ''):command_target("strSubcommand"):epilog(strEpilog)
+
+tParser:flag "--version":description "Show version info and exit. ":action(function()
+    require("flasher_version")
+    print(FLASHER_VERSION_STRING)
+    os.exit(0)
+end)
+
+-- 	flashfCommandFlashSelected
+local tParserCommandFlash = tParser:command('flash f', 'Flash a file to the netX'):target('fCommandFlashSelected')
+-- required_args = {"b", "u", "cs", "s", "f"},
+addFilePathArg(tParserCommandFlash)

```

Comment:
This could be written in a more compact way, e.g.
addArgs(tParserCommandFlash, {"b", "u", "cs", "s", "f", "p", "t", "jf", "jr"})